app/feature/feature-profile/src/main/kotlin/com/hedvig/android/feature/profile/tab/ProfileDestination.kt

```diff
@@ -214,6 +218,73 @@ private fun ProfileScreen(
   }
 }
 
+@Composable
+private fun ColumnScope.ProfileRowsWithPlaceholders(
+  profileUiState: ProfileUiState,
+  showMyInfo: () -> Unit,
+  showSettings: () -> Unit,
+  showAboutApp: () -> Unit,
+  navigateToEurobonus: () -> Unit,
+  navigateToTravelCertificate: () -> Unit,
+) {
+  AnimatedContent(
+    targetState = profileUiState.isLoading,
+    transitionSpec = { fadeIn() togetherWith fadeOut() },
+  ) { isLoading ->
+    Column(Modifier.fillMaxSize()) {
+      when (isLoading) {
+        true -> {
+          ProfileItemRowsPlaceholders()
+        }
+
+        false -> {
+          ProfileItemRows(
+            profileUiState = profileUiState,
+            showMyInfo = showMyInfo,
+            showSettings = showSettings,
+            showAboutApp = showAboutApp,
+            navigateToEurobonus = navigateToEurobonus,
+            navigateToTravelCertificate = navigateToTravelCertificate,
+          )
+        }
+      }
+    }
+  }
+}
+
+@Composable
+private fun ColumnScope.ProfileItemRowsPlaceholders() {
+  ProfileRowPlaceholder(stringResource(R.string.PROFILE_MY_INFO_ROW_TITLE))
+  ProfileRowPlaceholder(stringResource(R.string.PROFILE_ROW_TRAVEL_CERTIFICATE))
+  ProfileRowPlaceholder(stringResource(R.string.PROFILE_ABOUT_ROW))
+  ProfileRowPlaceholder(stringResource(R.string.profile_appSettingsSection_row_headline))
+}
+
+@Composable
+private fun ProfileRowPlaceholder(title: String) {
```

Comment:
This function and 
`private fun ProfileRow(title: String, icon: ImageVector, onClick: () -> Unit, modifier: Modifier = Modifier) {`
function each having their own UI logic means that they can also go out of sync quite easily. Could you instead look into providing another private composable in this file, which will contain the core logic of how the UI should be laid out. It may need to have some composable slots in there, I don't have something specific in mind unless I sit down and try to do it myself. And then have the placeholder and the non-placeholder composables call *that* instead. This way there will be better guarantees that these two do not grow out of sync.

In general as we try to do this more and more I am sure we will pick up better patters on how to do this without complicating our code too much and without making it possible to have the placeholder and non-placeholder composables go out of sync.